### PR TITLE
Add a section about the download endpoint

### PR DESCRIPTION
**Pull request by clementprdhomme**
This PR adds a simple section to document the `download` endpoint.

<p align="center">
Screenshot of the new content
</p>

### CODE REVIEW

File: source/includes/api/_query.md [hunk 1]
```diff
@@ -162,3 +162,20 @@ You can delimit an area of interest by providing a geostore id as a parameter:
 ```shell
 curl -i -XGET http\://api.resourcewatch.org/v1/query/b99c5f5e-00c6-452e-877c-ced2b9f0b393\?sql\=SELECT\ \*\ from\ nexgddp-historical-ACCESS1_0-prmaxday\ \ where\ year\ between\ 1960\ and\ 1962&geostore\=0279093c278a64f4c3348ff63e4cfce0
 ```
+
+## Download
+
+You can download the result of a query using the `download` endpoint. You can specify which format you want to use (JSON or CSV) for all the dataset providers except Google Earth Engine, for which you can only download a JSON file.
```
Comment (tiagojsag) [inline source/includes/api/_query.md]: Could I ask you to add a line about the 2 ways to download data:
- /download?sql=select * from <dataset.id>
- /download/<dataset.id>?sql=select *

File: source/includes/api/_query.md [hunk 2]
```diff
@@ -162,3 +162,20 @@ You can delimit an area of interest by providing a geostore id as a parameter:
...
+
+<aside class="notice">
```
Comment (tiagojsag) [inline source/includes/api/_query.md:196]: Can you add a warning with the list of dataset types that support download. There are certain types that don't support it

Comment (clementprdhomme) [reply source/includes/api/_query.md:196]: When I reviewed it, every provider supported it (I checked the adapters). Only Google Earth Engine isn't able to provide a CSV.

I checked the code for: document, arcgis, carto, bigquery and earth engine. Am I missing something?

Comment (tiagojsag) [reply source/includes/api/_query.md:196]: there are more adapters than those: https://github.com/resource-watch/dataset/blob/master/app/src/app.constants.js#L4

But yes, AFAIK those are the only ones that support download

Comment (clementprdhomme) [reply source/includes/api/_query.md:196]: Thanks! Let me update that then!

**Comment by clementprdhomme** [comment]
@tiagojsag I think I've addressed your comments. What do you think?

**Pull request merged**